gcp/databases/cloudsql/postgresql/main.tf

```diff
@@ -0,0 +1,297 @@
+        query_string_length     = lookup(insights_config.value, "query_string_length", 1024)
+        record_application_tags = lookup(insights_config.value, "record_application_tags", false)
+        record_client_address   = lookup(insights_config.value, "record_client_address", false)
+      }
+    }
+
+    disk_autoresize       = var.disk_autoresize
+    disk_autoresize_limit = var.disk_autoresize_limit
+    disk_size             = var.disk_size
+    disk_type             = var.disk_type
+    pricing_plan          = var.pricing_plan
+    dynamic database_flags {
+      for_each = var.database_flags
+      content {
+        name  = lookup(database_flags.value, "name", null)
+        value = lookup(database_flags.value, "value", null)
+      }
+    }
+
+    user_labels = var.user_labels
+
+    location_preference {
+      zone = var.zone
+    }
+
+    maintenance_window {
+      day          = var.maintenance_window_day
+      hour         = var.maintenance_window_hour
+      update_track = var.maintenance_window_update_track
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      settings[0].disk_size
+    ]
+  }
+
+  timeouts {
+    create = var.create_timeout
+    update = var.update_timeout
+    delete = var.delete_timeout
+  }
+
+  depends_on = [null_resource.module_depends_on]
+}
+
+resource google_sql_database default {
+  count      = var.enable_default_db ? 1 : 0
+  name       = var.db_name
+  project    = var.project_id
+  instance   = google_sql_database_instance.default.name
+  charset    = var.db_charset
+  collation  = var.db_collation
+  depends_on = [null_resource.module_depends_on, google_sql_database_instance.default]
+}
+
+resource google_sql_database additional_databases {
+  for_each   = local.databases
+  project    = var.project_id
+  name       = each.value.name
+  charset    = lookup(each.value, "charset", null)
+  collation  = lookup(each.value, "collation", null)
+  instance   = google_sql_database_instance.default.name
+  depends_on = [null_resource.module_depends_on, google_sql_database_instance.default]
+}
+
+resource random_password user-password {
+  keepers = {
+    name = google_sql_database_instance.default.name
+  }
+
+  length     = 32
+  special    = false
+  depends_on = [null_resource.module_depends_on, google_sql_database_instance.default]
+}
+
+resource google_secret_manager_secret user-password {
+  project = var.project_id
+  secret_id   = "user-password"
+  replication {
+    user_managed {
+      replicas {
+        location = var.region
+      }
+    }
+  }
+
+  depends_on = [null_resource.module_depends_on, random_password.user-password]
+}
+
+resource google_secret_manager_secret_version user-password {
+  secret = google_secret_manager_secret.user-password.id
+  secret_data = random_password.user-password.result
+
+  depends_on = [null_resource.module_depends_on, google_secret_manager_secret.user-password]
+}
+
+resource random_password additional_passwords {
```

Comment:
Ah. This remains static and doesn't randomise each run? Does this get stored in the state file?